File: SERVICES_FE/inventory/src/model/SubCategory.ts
import { Category } from "./Category";

export interface SubCategory {
  id: number;

  name: string;

  active: boolean;

  createdOn: String;

  updatedOn: String;

  category: Category;
}

export interface SubCategoryLabel {
  id: number;

  name: string;
}

File: SERVICES_BE/inventory-service/src/main/java/io/active/kart/base/entity/SubCategory.java
package io.active.kart.base.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "sub_category")
public class SubCategory extends BaseEntity {

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @Override
    public String toString() {
        return " SubCategory < " + super.toString() + category + " > ";
    }

}

